src/src/PluginStructs/P109_data_struct.cpp

```diff
@@ -81,14 +81,14 @@ bool P109_data_struct::plugin_init(struct EventStruct *event) {
     delete _display;
     _display = nullptr;
   }
-  _taskIndex       = event->TaskIndex;
-  _varIndex        = event->BaseVarIndex;
-  _relaypin        = P109_CONFIG_RELAYPIN;
-  _relayInverted   = P109_GET_RELAY_INVERT;
-  _buttonInverted[0]   = P109_GET_BUTTON1_INVERT;
-  _buttonInverted[1]   = P109_GET_BUTTON2_INVERT;
-  _buttonInverted[2]   = P109_GET_BUTTON3_INVERT;
-  _setpointTimeout = P109_CONFIG_SETPOINT_DELAY - P109_SETPOINT_OFFSET;
+  _taskIndex            = event->TaskIndex;
+  _varIndex             = event->BaseVarIndex;
+  _relaypin             = P109_CONFIG_RELAYPIN;
+  _relayInverted        = P109_GET_RELAY_INVERT;
+  _buttonNoIntPullup[0] = P109_GET_BUTTON1_NO_INTPULLUP;
+  _buttonNoIntPullup[1] = P109_GET_BUTTON2_NO_INTPULLUP;
+  _buttonNoIntPullup[2] = P109_GET_BUTTON3_NO_INTPULLUP;
```

Comment:
Not sure if this change of name removes any confusion...

Reply:
Well, I thought this: by default (including in this plugin) the buttons are IntPullUp. Therefore, if the button is PullUp or PullDown it is NO_INTPULLUP. Of course I accept your suggestions )

Reply:
Then let it be INT_PULL_UP (and the function logic is reverse to the current one). OK?

Reply:
Sure, but then you must make sure the defaults are dealt with.
When you don't set the defaults, the value should be assumed 0.

There is a specific `PLUGIN_CALL_....` for loading the defaults which will only be called when you set the plugin to a task.